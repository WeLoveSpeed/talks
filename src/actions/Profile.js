import {
  requestPeople
} from '../services/GoogleAPI';

export const LOAD_PROFILE = 'LOAD_PROFILE';
export const FETCHED_PROFILE = 'FETCHED_PROFILE';

export const loadProfile = result => dispatch => {
  dispatch({
    type: LOAD_PROFILE
  });

  requestPeople()
  .then(response => {
    const result = response.result;
    let profile = {};
    if (result.names) {
      profile.name = result.names[0].displayName;
    }
    if (result.emailAddresses) {
      profile.email = result.emailAddresses[0].value;
    }

    if (result.photos) {
      profile.img = result.photos[0].url;
    }
    return dispatch({
      type: FETCHED_PROFILE,
      profile: profile
    })
  }, error => this.handleError(error))
}