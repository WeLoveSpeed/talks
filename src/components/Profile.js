import React, { Component } from "react";
import Avatar from "material-ui/Avatar";
import { ListItem } from "material-ui/List";
import IconButton from "material-ui/IconButton";
import IconMenu from "material-ui/IconMenu";
import MenuItem from "material-ui/MenuItem";
import MoreVertIcon from "material-ui/svg-icons/navigation/more-vert";

class Profile extends Component {
  render() {
    const { profile, signout } = this.props;

    if (!profile) {
      return null;
    }

    const styleProfile = {
      color: "white",
      paddingTop: !profile.email ? 24 : 14,
      paddingBottom: !profile.email ? 4 : 14,
      paddingRight: 42
    };

    const iconButtonElement = (
      <IconButton
        touch={true}
        tooltipPosition="top-right"
        style={{ paddingRight: 0, width: 32 }}
      >
        <MoreVertIcon color="white" />
      </IconButton>
    );

    const rightIconMenu = (
      <IconMenu iconButtonElement={iconButtonElement}>
        <MenuItem onClick={signout}>DÃ©connexion</MenuItem>
      </IconMenu>
    );

    return (
      <ListItem
        disabled={true}
        innerDivStyle={styleProfile}
        primaryText={profile.name || ""}
        secondaryText={profile.email || " "}
        leftAvatar={
          (
            <Avatar
              style={{ top: 12 }}
              src={profile.img}
              backgroundColor="white"
            />
          )
        }
        rightIconButton={rightIconMenu}
      />
    );
  }
}

export default Profile;
