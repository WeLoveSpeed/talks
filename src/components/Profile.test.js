import React, { Component } from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Profile from "./Profile";

const fn = function() {
  console.log("signout");
};

describe("Profile component", () => {
  it("should renders correctly", () => {
    const component = shallow(
      <Profile
        profile={{ name: "Alex", email: "test@test.com" }}
        signout={fn}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render null", () => {
    const component = shallow(<Profile signout={fn} />);
    expect(component).toMatchSnapshot();
  });
});
