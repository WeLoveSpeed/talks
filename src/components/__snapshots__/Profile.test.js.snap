exports[`Profile component should render null 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": null,
  "nodes": Array [
    null,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Profile
        signout={[Function]} />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Profile {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "signout": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": null,
        "_debugID": 4,
        "_renderedOutput": null,
      },
      "_renderedNodeType": 2,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Profile
    signout={[Function]} />,
}
`;

exports[`Profile component should renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ListItem
    autoGenerateNestedIndicator={true}
    disableKeyboardFocus={false}
    disabled={true}
    initiallyOpen={false}
    innerDivStyle={
        Object {
            "color": "white",
            "paddingBottom": 14,
            "paddingRight": 42,
            "paddingTop": 14,
          }
    }
    insetChildren={false}
    leftAvatar={
        <Avatar
            backgroundColor="white"
            size={40}
            src={undefined}
            style={
                Object {
                    "top": 12,
                  }
            } />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    primaryText="Boris"
    primaryTogglesNestedList={false}
    rightIconButton={
        <IconMenu
            anchorOrigin={
                Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
            }
            animated={true}
            iconButtonElement={
                <IconButton
                    disableTouchRipple={false}
                    disabled={false}
                    iconStyle={Object {}}
                    style={
                        Object {
                            "paddingRight": 0,
                            "width": 32,
                          }
                    }
                    tooltipPosition="top-right"
                    touch={true}>
                    <NavigationMoreVert
                        color="white" />
                </IconButton>
            }
            multiple={false}
            onItemTouchTap={[Function]}
            onKeyboardFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onRequestChange={[Function]}
            onTouchTap={[Function]}
            open={null}
            targetOrigin={
                Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
            }
            touchTapCloseDelay={200}
            useLayerForClickAway={false}>
            <MenuItem
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                onClick={[Function]}>
                Déconnexion
            </MenuItem>
        </IconMenu>
    }
    secondaryText="test@test.com"
    secondaryTextLines={1} />,
  "nodes": Array [
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={true}
      initiallyOpen={false}
      innerDivStyle={
            Object {
                  "color": "white",
                  "paddingBottom": 14,
                  "paddingRight": 42,
                  "paddingTop": 14,
                }
      }
      insetChildren={false}
      leftAvatar={
            <Avatar
                  backgroundColor="white"
                  size={40}
                  src={undefined}
                  style={
                        Object {
                              "top": 12,
                            }
                  } />
      }
      nestedItems={Array []}
      nestedLevel={0}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Boris"
      primaryTogglesNestedList={false}
      rightIconButton={
            <IconMenu
                  anchorOrigin={
                        Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
                  }
                  animated={true}
                  iconButtonElement={
                        <IconButton
                              disableTouchRipple={false}
                              disabled={false}
                              iconStyle={Object {}}
                              style={
                                    Object {
                                          "paddingRight": 0,
                                          "width": 32,
                                        }
                              }
                              tooltipPosition="top-right"
                              touch={true}>
                              <NavigationMoreVert
                                    color="white" />
                        </IconButton>
                  }
                  multiple={false}
                  onItemTouchTap={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onRequestChange={[Function]}
                  onTouchTap={[Function]}
                  open={null}
                  targetOrigin={
                        Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
                  }
                  touchTapCloseDelay={200}
                  useLayerForClickAway={false}>
                  <MenuItem
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        onClick={[Function]}>
                        Déconnexion
                  </MenuItem>
            </IconMenu>
      }
      secondaryText="test@test.com"
      secondaryTextLines={1} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Profile
        profile={
                Object {
                        "email": "test@test.com",
                        "name": "Boris",
                      }
        }
        signout={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Profile {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "profile": Object {
            "email": "test@test.com",
            "name": "Boris",
          },
          "signout": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ListItem
          autoGenerateNestedIndicator={true}
          disableKeyboardFocus={false}
          disabled={true}
          initiallyOpen={false}
          innerDivStyle={
                    Object {
                              "color": "white",
                              "paddingBottom": 14,
                              "paddingRight": 42,
                              "paddingTop": 14,
                            }
          }
          insetChildren={false}
          leftAvatar={
                    <Avatar
                              backgroundColor="white"
                              size={40}
                              src={undefined}
                              style={
                                        Object {
                                                  "top": 12,
                                                }
                              } />
          }
          nestedItems={Array []}
          nestedLevel={0}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="Boris"
          primaryTogglesNestedList={false}
          rightIconButton={
                    <IconMenu
                              anchorOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                              animated={true}
                              iconButtonElement={
                                        <IconButton
                                                  disableTouchRipple={false}
                                                  disabled={false}
                                                  iconStyle={Object {}}
                                                  style={
                                                            Object {
                                                                      "paddingRight": 0,
                                                                      "width": 32,
                                                                    }
                                                  }
                                                  tooltipPosition="top-right"
                                                  touch={true}>
                                                  <NavigationMoreVert
                                                            color="white" />
                                        </IconButton>
                              }
                              multiple={false}
                              onItemTouchTap={[Function]}
                              onKeyboardFocus={[Function]}
                              onMouseDown={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onRequestChange={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              targetOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                              touchTapCloseDelay={200}
                              useLayerForClickAway={false}>
                              <MenuItem
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        onClick={[Function]}>
                                        Déconnexion
                              </MenuItem>
                    </IconMenu>
          }
          secondaryText="test@test.com"
          secondaryTextLines={1} />,
        "_debugID": 2,
        "_renderedOutput": <ListItem
          autoGenerateNestedIndicator={true}
          disableKeyboardFocus={false}
          disabled={true}
          initiallyOpen={false}
          innerDivStyle={
                    Object {
                              "color": "white",
                              "paddingBottom": 14,
                              "paddingRight": 42,
                              "paddingTop": 14,
                            }
          }
          insetChildren={false}
          leftAvatar={
                    <Avatar
                              backgroundColor="white"
                              size={40}
                              src={undefined}
                              style={
                                        Object {
                                                  "top": 12,
                                                }
                              } />
          }
          nestedItems={Array []}
          nestedLevel={0}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="Boris"
          primaryTogglesNestedList={false}
          rightIconButton={
                    <IconMenu
                              anchorOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                              animated={true}
                              iconButtonElement={
                                        <IconButton
                                                  disableTouchRipple={false}
                                                  disabled={false}
                                                  iconStyle={Object {}}
                                                  style={
                                                            Object {
                                                                      "paddingRight": 0,
                                                                      "width": 32,
                                                                    }
                                                  }
                                                  tooltipPosition="top-right"
                                                  touch={true}>
                                                  <NavigationMoreVert
                                                            color="white" />
                                        </IconButton>
                              }
                              multiple={false}
                              onItemTouchTap={[Function]}
                              onKeyboardFocus={[Function]}
                              onMouseDown={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onRequestChange={[Function]}
                              onTouchTap={[Function]}
                              open={null}
                              targetOrigin={
                                        Object {
                                                  "horizontal": "left",
                                                  "vertical": "top",
                                                }
                              }
                              touchTapCloseDelay={200}
                              useLayerForClickAway={false}>
                              <MenuItem
                                        checked={false}
                                        desktop={false}
                                        disabled={false}
                                        focusState="none"
                                        insetChildren={false}
                                        onClick={[Function]}>
                                        Déconnexion
                              </MenuItem>
                    </IconMenu>
          }
          secondaryText="test@test.com"
          secondaryTextLines={1} />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Profile
    profile={
        Object {
            "email": "test@test.com",
            "name": "Boris",
          }
    }
    signout={[Function]} />,
}
`;
